mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -82,9 +82,17 @@
     String DEFAULT = "root";
     String ERROR = "error";
     String TRANSPORT = "transport";
-    String SMTP_AUTH_USER_ATTRIBUTE_NAME = "org.apache.james.SMTPAuthUser";
     String SENT_BY_MAILET = "org.apache.james.SentByMailet";
+
+    @Deprecated
+    String SMTP_AUTH_USER_ATTRIBUTE_NAME = "org.apache.james.SMTPAuthUser";
+    @SuppressWarnings("deprecation")
+    AttributeName SMTP_AUTH_USER = AttributeName.of(SMTP_AUTH_USER_ATTRIBUTE_NAME);
+
+    @Deprecated
     String MAILET_ERROR_ATTRIBUTE_NAME = "org.apache.james.MailetError";
+    @SuppressWarnings("deprecation")
+    AttributeName MAILET_ERROR = AttributeName.of(MAILET_ERROR_ATTRIBUTE_NAME);
```

Comment:
You can get ride of the `@SuppressWarnings("deprecation")` by reversing the declaration:

```
AttributeName MAILET_ERROR = AttributeName.of("org.apache.james.MailetError");

@Deprecated
String MAILET_ERROR_ATTRIBUTE_NAME = MAILET_ERROR.asString();
```